### PR TITLE
Class Modules (#690)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve Issue #690 

To be reviewed after 2.3 is released!!!

```python
from src.taipy.gui import Markdown, Gui, ClassModule


class TestModule(ClassModule):
    def __init__(self):
        self.x = 10
        self.y = 20
        self.page = Markdown(
            """
x = <|{x}|>

<|Set x to 30|button|on_action=set_x|>
"""
        )
    def set_x(self):
        print("hey")
        self.x = 30

    def on_change(self):
        pass


gui = Gui()
gui.add_page("test", TestModule)
gui.run(run_browser=False)
```

### CODE REVIEW

File: src/taipy/gui/page.py [hunk 1]
```diff
@@ -85,10 +87,27 @@ def _get_content_detail(self, gui) -> str:
...
+            self._class_locals
+            if self._class_locals != {}
+            else None
+            if self._frame is None
+            else _filter_locals(self._frame.f_locals)
+        )
+
+    def _set_class_module(self, name: str, locals_: t.Dict[str, t.Any]) -> None:
+        self._class_module_name = name
+        self._class_locals = locals_
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/page.py]: What happens if class has no other member than page?

**Review by FredLL-Avaiga — APPROVED** [review]
simple & effective 👍

File: src/taipy/gui/page.py (hunk 1 above)
Comment (dinhlongviolin1) [reply src/taipy/gui/page.py]: then it should be an empty locals, as it should be

File: src/taipy/gui/__init__.py [hunk 2]
```diff
@@ -70,6 +70,7 @@
 
 from importlib.util import find_spec
 
+from .class_module import ClassModule
```
Comment (FabienLelaquais) [inline src/taipy/gui/__init__.py:74]: My proposal - to be discussed
`from .base_page import BasePage`

**Review by FabienLelaquais — APPROVED** [review]
I totally love it.
So simple! So neat!

**Comment by dinhlongviolin1** [comment]
This one is obsolete, replaced by PR #816

**Pull request closed without merging**